ntbs-service/Pages/ClinicalDetails/Edit.cshtml.cs

```diff
@@ -0,0 +1,128 @@
+        public FormattedDate FormattedPresentationDate { get; set; }
+        [BindProperty]
+        public FormattedDate FormattedDiagnosisDate { get; set; }
+        [BindProperty]
+        public FormattedDate FormattedTreatmentDate { get; set; }
+        [BindProperty]
+        public FormattedDate FormattedDeathDate { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            var notification = await service.GetNotificationAsync(id);
+            if (notification == null)
+            {
+                return NotFound();
+            }
+
+            ClinicalDetails = notification.ClinicalDetails;
+            NotificationId = notification.NotificationId;
+
+            if (ClinicalDetails == null) {
+                ClinicalDetails = new ClinicalDetails();
+            }
+
+            var notificationSites = notification.NotificationSites;
+            OtherSite = notificationSites.FirstOrDefault(ns => ns.SiteId == (int)SiteId.OTHER);
+            Sites = (await _context.GetAllSitesAsync()).ToList();
+            NotificationSiteMap = CreateNotificationSiteMap(notificationSites, Sites);
+
+            PatientBirthYear = notification.PatientDetails.Dob?.Year;
+
+            FormattedSymptomDate = ClinicalDetails.SymptomStartDate.ConvertToFormattedDate();
+            FormattedPresentationDate = ClinicalDetails.PresentationDate.ConvertToFormattedDate();
+            FormattedDiagnosisDate = ClinicalDetails.DiagnosisDate.ConvertToFormattedDate();
+            FormattedTreatmentDate = ClinicalDetails.TreatmentStartDate.ConvertToFormattedDate();
+            FormattedDeathDate = ClinicalDetails.DeathDate.ConvertToFormattedDate();
+
+            return Page();
+        }
+
+        private Dictionary<SiteId, bool> CreateNotificationSiteMap(IEnumerable<NotificationSite> notificationSites, IEnumerable<Site> sites)
+        {
+            var dict = new Dictionary<SiteId, bool>();
+            foreach (var site in sites)
+            {
+                dict.Add((SiteId)site.SiteId, notificationSites.FirstOrDefault(ns => ns.SiteId == site.SiteId) != null);
+            }
+            return dict;
+        }
+
+        public async Task<IActionResult> OnPostAsync(int? id)
+        {
+            SetAndValidateDate(ClinicalDetails, nameof(ClinicalDetails.SymptomStartDate), FormattedSymptomDate);
+            SetAndValidateDate(ClinicalDetails, nameof(ClinicalDetails.PresentationDate), FormattedPresentationDate);
+            SetAndValidateDate(ClinicalDetails, nameof(ClinicalDetails.DiagnosisDate), FormattedDiagnosisDate);
+            SetAndValidateDate(ClinicalDetails, nameof(ClinicalDetails.TreatmentStartDate), FormattedTreatmentDate);
+            SetAndValidateDate(ClinicalDetails, nameof(ClinicalDetails.DeathDate), FormattedDeathDate);
+            ValidateYearAgainstOtherYear(ClinicalDetails, nameof(ClinicalDetails.BCGVaccinationYear), ClinicalDetails.BCGVaccinationYear, PatientBirthYear);

```

Comment:
Need to do some manual validation here as we're comparing to year of birth which is defined on a different model (PatientDetails) - I added it to the ValidationModel subclass and kept it fairly general in case we need to compare two years in this way elsewhere.

Reply:
nice!